_sources/structs/mixed_up_code.rst

```diff
@@ -0,0 +1,236 @@
+
+.. parsonsprob:: structs_mixed_up_code_6
+    :numbered: left
+    :adaptive:
+
+    Let's write the code for the ``printAddress`` function. ``printAddress`` takes
+    an ``Employee`` as a parameter and should print out the information of the employee in the 
+    following format: ``name (id) lives at houseNumber in state, postalAddress``.
+    Put the necessary blocks of code in the correct order.
+    -----
+    void printAddress (const Employee &e) {
+    =====
+    string printAddress (Employee &e) {  #paired 
+    =====
+        cout << e.name << " (" << e.id << ") lives at ";
+    =====
+        cout << e.address.name << " (" << e.address.id << ") lives at ";  #distractor
+    =====
+        cout << e.name << "(" << e.address.id << ") lives at";  #distractor
+    =====
+        cout << e.address.houseNumber << " in " << e.address.state << ", " << e.address.postalAddress << endl;
+    =====
+        cout << e.houseNumber << " in " << e.state << ", " << e.postalAddress << endl;  #distractor
+    =====
+    }
+
+.. parsonsprob:: structs_mixed_up_code_7_re
+    :numbered: left
+    :adaptive:
+
+    Sometimes employees will move around and thus we'll need to update their addresses.
+    Let's write the code for the ``updateAddress`` function. updateAddress takes an
+    ``Employee`` and a new ``Address`` as parameters and sets the employee's address to the new address.
+    Put the necessary blocks of code in the correct order.
+    -----
+    void updateAddress (Employee &e, Address &a) {
+    =====
+    void updateAddress (Employee e, Address a) {  #distractor
+    =====
+        e.address = a;
+    =====
+        e.address.state = a.state;  #distractor
+    =====
+        e.address.houseNumber = a.houseNumber;  #distractor
+    =====
+        e.address.postalAddress = a.postalAddress;  #distractor
+    =====
+    }
+    =====
+    };  #distractor
+
+.. parsonsprob:: structs_mixed_up_code_8_re
+    :numbered: left
+    :adaptive:
+
+    Let's write the code for the ``storeEmployeeData`` function. ``storeEmployeeData`` should be a void function that takes an empty ``Employee`` as an input.
+    ``storeEmployeeData`` prompts the user for information regarding their ``name``, ``houseNumber``, ``state``, and ``postalAddress`` in that order.
+    It then stores the user's information in the inputted empty ``Employee`` structure. Put the necessary blocks of code in the correct order.
+    -----
+    void storeEmployeeData (Employee &e) {
+    =====
+        cout << "What is your full name? ";
+    =====
+        cin >> e.name;
+    =====
+        cout << "What is your house number? ";
+    =====
+        cin >> e.address.houseNumber;
+    =====
+        cin >> e.houseNumber;  #paired
+    =====
+        cout << "What state do you live in? ";
+    =====
+        cin >> e.address.state;
+    =====
+        cout << "What is your postal address? ";
+    =====
+        cin >> e.address.postalAddress;
+    =====
+        return Employee e;  #paired
```

Comment:
Why is this one paired with the previous line? It seems like maybe it shouldn't be paired

Reply:
Yes, this should have been a #distractor, thanks. Fixed now!